### PR TITLE
fix: upload only known build output directories

### LINKED ISSUE #270: Upload only necessary build files
With the inclusion of sentry-native for Windows builds (https://github.com/getsentry/sentry-dart/pull/2286) the plugin exhibits an issue.

When run, the CLI is trying to upload everything in the current directory but there are some intentionally broken debug files inside one of the sentry-native dependencies' test files so the upload fails with "Invalid checksum" (and no details because CLI logs are not piped...)

This is what it looks like:

```
entry CLI downloaded successfully.

retrieving config from pubspec.yaml
☑ reading config values                                                             
☑ validating config values                                                             
[❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚❚] 18% uploading debug symbols..                                
INFO    2024-10-15 20:06:18.197891599 +02:00 sentry-cli was invoked with the following command line: "/home/ivan/dev/sentry-dart/flutter/example/.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli" "--log-level" "info" "debug-files" "upload" "--org" "sentry-sdks" "--project" "sentry-flutter" "--include-sources" "/home/ivan/dev/sentry-dart/flutter/example" "--wait"
  INFO    2024-10-15 20:06:18.197961939 +02:00 Issuing a command for Organization: sentry-sdks Project: sentry-flutter

error: Invalid checksum

exitCode: 1
```

while when you launch sentry-cli manually, you can see the context:

```
  DEBUG   2024-10-15 22:02:55.368911765 +02:00 finished zip archive /home/ivan/dev/sentry-dart/flutter/example/linux/flutter/ephemeral/.plugin_symlinks/sentry_flutter/example/build/linux/x64/release/_deps/sentry-native-src/external/crashpad/third_party/zlib/zlib/google/test/data/test_nocompress.zip
  DEBUG   2024-10-15 22:02:55.368959756 +02:00 searching zip archive /home/ivan/dev/sentry-dart/flutter/example/linux/flutter/ephemeral/.plugin_symlinks/sentry_flutter/example/build/linux/x64/release/_deps/sentry-native-src/external/crashpad/third_party/zlib/zlib/google/test/data/Wrong CRC.zip
error: Invalid checksum
```

## Proposed solution

Uploading everything in the current directory is not really necessary - we only need to upload debug files for the recent build. The problem is that we don't have a sure way to know what was built (see #94). Instead, we can look for the build output files based on the path & name - we know they're named a certain way for each platform build and we can search the current directory recursively until we find the right structure and upload only what's necessary. If we can't locate the build according to the rules in the plugin, we can fall back to the current behaviour of uploading eveything.

### Benefits

- faster debug symbol uploads (unnecessary files are not uploaded)
- less space occupied on Sentry
- avoid potential issues with broken files

### Problems

- we may miss some of the debug files if the logic that locates them is incomplete

### PR DESCRIPTION
**Pull request by vaind**
## :scroll: Description

- change build dir default value
- add symbols path config
- enumarate known build dir paths
- update tests to test all possible targets of `flutter build`

## :bulb: Motivation and Context

fixes #270 

## :pencil: Checklist

- [x] depends on #279 
- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [ ] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: example/README.md [hunk 1]
```diff
@@ -37,9 +37,15 @@ sentry:
   # default 'warning'
   log_level: error
 
-  # default to build/web
+  # default 'build'
+  #build_path: ...
+
+  # default 'web'
```
Comment (denrase) [inline example/README.md:43]: Should we also mention here that this is relative to `build_path`?

File: lib/sentry_dart_plugin.dart [hunk 2]
```diff
@@ -75,15 +78,86 @@ class SentryDartPlugin {
...
+    }
+
+    for (final path in await _enumerateSymbolFiles()) {
+      await _executeAndLog('Failed to upload symbols', [...params, path]);
+    }
 
     Log.taskCompleted(taskName);
   }
 
+  Stream<String> _enumerateDebugSymbolPaths() async* {
```
Comment (denrase) [inline lib/sentry_dart_plugin.dart:98]: Any particular reason why this is a stream when we are "just" returning a list of directories?

File: lib/src/configuration.dart [hunk 3]
```diff
@@ -67,9 +66,12 @@ class Configuration {
   /// The Apps name, defaults to [name] from pubspec
   late String name;
 
-  /// the Web Build folder, defaults to build/web
+  /// the Web Build folder, defaults to `web`
```
Comment (denrase) [inline lib/src/configuration.dart:69]: Could also indicate that this is relative to build, if it indeed is.

File: test/integration_test.dart [hunk 4]
```diff
@@ -257,3 +243,22 @@ final _serverPort =
   socket.close();
   return port;
 });
+
+class _ProcessStreamCollector {
+  final StringBuffer _output = StringBuffer();
```
Comment (denrase) [inline test/integration_test.dart:248]: nit: no need for type annotation here

**Review by denrase — APPROVED** [review]
LGTM, just some minor comments/questions. 🙇‍♂️

**Review by buenaflor — APPROVED** [review]
👍

File: lib/src/configuration.dart (hunk 3 above)
Comment (vaind) [reply lib/src/configuration.dart:69]: done

File: test/integration_test.dart (hunk 4 above)
Comment (vaind) [reply test/integration_test.dart:248]: done

File: example/README.md (hunk 1 above)
Comment (vaind) [reply example/README.md:43]: done

File: lib/sentry_dart_plugin.dart (hunk 2 above)
Comment (vaind) [reply lib/sentry_dart_plugin.dart:98]: none, just convenient syntax

**Pull request merged**